providers/oidc.go

```diff
@@ -109,19 +152,14 @@ func (p *OIDCProvider) redeemRefreshToken(ctx context.Context, s *sessions.Sessi
 		return fmt.Errorf("failed to get token: %v", err)
 	}
 
-	// in the token refresh response the id_token is optional
```

Comment:
The refresh vs redeem verified IDToken requirements were moved into `createSession` to reduce redundant code and logic branches.